distribution/std-lib/Base/src/Data/Number/Extensions.enso

```diff
@@ -98,7 +102,17 @@ Number.to_json = Json.Number this
 
 ## Parses a textual representation of a decimal into a decimal number.
    Returns `Nothing` if the text does not represent a valid decimal.
-Decimal.parse : Text -> Decimal | Nothing
+Decimal.parse : Text -> Decimal ! Number_Parse_Error
 Decimal.parse text =
-    Panic.recover (Double.parseDouble [text]) . catch (_ -> Nothing)
-
+    Panic.recover (Double.parseDouble [text]) . catch (_ -> Error.throw Number_Parse_Error)
+
+## Parses a textual representation of an integer into an integer number.
+   Returns `Nothing` if the text does not represent a valid integer.
+Integer.parse : Text -> Integer ! Number_Parse_Error
+Integer.parse text =
+    Panic.recover (Java_Integer.parseInt [text]) . catch (_ -> Error.throw Number_Parse_Error)
+
+## Parses a textual representation of a number into an integer or decimal number.
+   Returns `Nothing` if the text does not represent a valid number.
+Number.parse : Text -> Number ! Number_Parse_Error
+Number.parse text = Integer.parse text . catch (_ -> Decimal.parse text)
```

Comment:
Once you add descriptions to errors in this case it won't make sense here. You'll need to explicitly set the error message for this method failing.